Compile error with -Weffc++

On current master I get the following compile error:
```
../src/lib/matrix/matrix/Vector.hpp:20:7: error: base class 'class matrix::Matrix<float, 3u, 1u>' has a non-virtual destructor [-Werror=effc++]
 class Vector : public Matrix<Type, M, 1>
```

This is a false alarm and got introduced with https://github.com/PX4/Matrix/commit/66e1b406b8a68422135c365d42b987a64c9ddf8d

In general I agree with Scott Meyer's recommondations, but I'm starting to doubt the usefulness of the GCC flag. Also since we have clang-tidy that does similar things. @dagar what do you think?

--- a/cmake/common/px4_base.cmake
+++ b/cmake/common/px4_base.cmake
@@ -820,7 +820,6 @@ function(px4_add_common_flags)
 
 	set(cxx_warnings
 		-Wno-missing-field-initializers
-		#-Weffc++
 		)
 
 	set(cxx_compile_flags

--- a/src/drivers/bmi160/CMakeLists.txt
+++ b/src/drivers/bmi160/CMakeLists.txt
@@ -35,7 +35,6 @@ px4_add_module(
 	MAIN bmi160
 	STACK_MAIN 1200
 	COMPILE_FLAGS
-		-Weffc++
 	SRCS
 		bmi160.cpp
 		bmi160_gyro.cpp

--- a/src/drivers/hmc5883/CMakeLists.txt
+++ b/src/drivers/hmc5883/CMakeLists.txt
@@ -35,7 +35,6 @@ px4_add_module(
 	MAIN hmc5883
 	STACK_MAIN 1200
 	COMPILE_FLAGS
-		-Weffc++
 	SRCS
 		hmc5883_i2c.cpp
 		hmc5883_spi.cpp

--- a/src/drivers/ist8310/CMakeLists.txt
+++ b/src/drivers/ist8310/CMakeLists.txt
@@ -35,7 +35,6 @@ px4_add_module(
 	MAIN ist8310
 	STACK_MAIN 1200
 	COMPILE_FLAGS
-		-Weffc++
 		-Os
 	SRCS
 		ist8310.cpp

--- a/src/drivers/l3gd20/CMakeLists.txt
+++ b/src/drivers/l3gd20/CMakeLists.txt
@@ -35,7 +35,6 @@ px4_add_module(
 	MAIN l3gd20
 	STACK_MAIN 1200
 	COMPILE_FLAGS
-		-Weffc++
 	SRCS
 		l3gd20.cpp
 	DEPENDS

--- a/src/drivers/lis3mdl/CMakeLists.txt
+++ b/src/drivers/lis3mdl/CMakeLists.txt
@@ -35,7 +35,6 @@ px4_add_module(
 	MAIN lis3mdl
 	STACK_MAIN 1200
 	COMPILE_FLAGS
-		-Weffc++
 	SRCS
 		lis3mdl_i2c.cpp
 		lis3mdl_spi.cpp

--- a/src/drivers/lps25h/CMakeLists.txt
+++ b/src/drivers/lps25h/CMakeLists.txt
@@ -35,7 +35,6 @@ px4_add_module(
 	MAIN lps25h
 	STACK_MAIN 1200
 	COMPILE_FLAGS
-		-Weffc++
 	SRCS
 		lps25h.cpp
 		lps25h_i2c.cpp

--- a/src/drivers/lsm303d/CMakeLists.txt
+++ b/src/drivers/lsm303d/CMakeLists.txt
@@ -35,7 +35,6 @@ px4_add_module(
 	MAIN lsm303d
 	STACK_MAIN 1200
 	COMPILE_FLAGS
-		-Weffc++
 	SRCS
 		lsm303d.cpp
 	DEPENDS

--- a/src/drivers/meas_airspeed/CMakeLists.txt
+++ b/src/drivers/meas_airspeed/CMakeLists.txt
@@ -35,7 +35,6 @@ px4_add_module(
 	MAIN meas_airspeed
 	STACK_MAIN 1200
 	COMPILE_FLAGS
-		-Weffc++
 	SRCS
 		meas_airspeed.cpp
 	DEPENDS

--- a/src/drivers/mpu6000/CMakeLists.txt
+++ b/src/drivers/mpu6000/CMakeLists.txt
@@ -35,7 +35,6 @@ px4_add_module(
 	MAIN mpu6000
 	STACK_MAIN 1400
 	COMPILE_FLAGS
-		-Weffc++
 	SRCS
 		mpu6000.cpp
 		mpu6000_i2c.cpp

--- a/src/drivers/mpu6500/CMakeLists.txt
+++ b/src/drivers/mpu6500/CMakeLists.txt
@@ -35,7 +35,6 @@ px4_add_module(
 	MAIN mpu6500
 	STACK_MAIN 1200
 	COMPILE_FLAGS
-		-Weffc++
 	SRCS
 		mpu6500.cpp
 	DEPENDS

--- a/src/drivers/mpu9250/CMakeLists.txt
+++ b/src/drivers/mpu9250/CMakeLists.txt
@@ -35,7 +35,6 @@ px4_add_module(
 	MAIN mpu9250
 	STACK_MAIN 1200
 	COMPILE_FLAGS
-		-Weffc++
 	SRCS
 		mpu9250.cpp
 		mpu9250_i2c.cpp

--- a/src/drivers/px4fmu/CMakeLists.txt
+++ b/src/drivers/px4fmu/CMakeLists.txt
@@ -35,7 +35,6 @@ px4_add_module(
 	MAIN fmu
 	STACK_MAIN 1200
 	COMPILE_FLAGS
-		-Weffc++
 	SRCS
 		fmu.cpp
 	DEPENDS

--- a/src/drivers/test_ppm/CMakeLists.txt
+++ b/src/drivers/test_ppm/CMakeLists.txt
@@ -35,7 +35,6 @@ px4_add_module(
 	MAIN test_ppm
 	STACK_MAIN 1200
 	COMPILE_FLAGS
-		-Weffc++
 	SRCS
 		test_ppm.cpp
 	DEPENDS

--- a/src/platforms/qurt/fc_addon/mpu_spi/CMakeLists.txt
+++ b/src/platforms/qurt/fc_addon/mpu_spi/CMakeLists.txt
@@ -53,7 +53,6 @@ px4_add_module(
 	MAIN mpu9x50
 	STACK_MAIN 1200
 	COMPILE_FLAGS
-		-Weffc++
 	SRCS
 		mpu9x50_main.cpp
 		mpu9x50_params.c

--- a/src/platforms/qurt/fc_addon/rc_receiver/CMakeLists.txt
+++ b/src/platforms/qurt/fc_addon/rc_receiver/CMakeLists.txt
@@ -53,7 +53,6 @@ px4_add_module(
         MAIN rc_receiver
 	STACK_MAIN 1200
         COMPILE_FLAGS
-                -Weffc++
         SRCS
                 rc_receiver_main.cpp
 		rc_receiver_params.c

--- a/src/platforms/qurt/fc_addon/uart_esc/CMakeLists.txt
+++ b/src/platforms/qurt/fc_addon/uart_esc/CMakeLists.txt
@@ -53,7 +53,6 @@ px4_add_module(
         MAIN uart_esc
 	STACK_MAIN 1200
         COMPILE_FLAGS
-                -Weffc++
         SRCS
                 uart_esc_main.cpp
 		uart_esc_params.c

